odoo/addons/base/models/avatar_mixin.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from base64 import b64encode
+from hashlib import sha512
+from odoo import models, fields, api
+
+from odoo.tools import html_escape, file_open
+
+
+class AvatarMixin(models.AbstractModel):
+    _name = 'avatar.mixin'
+    _inherit = ['image.mixin']
+    _description = "Avatar Mixin"
+
+    # all image fields are base64 encoded and PIL-supported
+    avatar_1920 = fields.Image("Avatar", max_width=1920, max_height=1920, compute="_compute_avatar_1920")
+    avatar_1024 = fields.Image("Avatar 1024", max_width=1024, max_height=1024, compute="_compute_avatar_1024")
+    avatar_512 = fields.Image("Avatar 512", max_width=512, max_height=512, compute="_compute_avatar_512")
+    avatar_256 = fields.Image("Avatar 256", max_width=256, max_height=256, compute="_compute_avatar_256")
+    avatar_128 = fields.Image("Avatar 128", max_width=128, max_height=128, compute="_compute_avatar_128")
+
+    def _compute_avatar(self, avatar_field, image_field):
+        for record in self:
+            avatar = record[image_field]
+            if not avatar:
+                if record[record._rec_name]:
+                    avatar = record._avatar_generate_svg()
+                else:
+                    avatar = record._avatar_get_placeholder()
+            record[avatar_field] = avatar
+
+    @api.depends(lambda self: [self._rec_name, 'image_1920'])
+    def _compute_avatar_1920(self):
+        self._compute_avatar('avatar_1920', 'image_1920')
+
+    @api.depends(lambda self: [self._rec_name, 'image_1024'])
+    def _compute_avatar_1024(self):
+        self._compute_avatar('avatar_1024', 'image_1024')
+
+    @api.depends(lambda self: [self._rec_name, 'image_512'])
+    def _compute_avatar_512(self):
+        self._compute_avatar('avatar_512', 'image_512')
+
+    @api.depends(lambda self: [self._rec_name, 'image_256'])
+    def _compute_avatar_256(self):
+        self._compute_avatar('avatar_256', 'image_256')
+
+    @api.depends(lambda self: [self._rec_name, 'image_128'])
+    def _compute_avatar_128(self):
+        self._compute_avatar('avatar_128', 'image_128')
+
+    def _avatar_get_seed(self):
+        return self[self._rec_name] + str(self.create_date.timestamp())
+
+    def _avatar_generate_svg(self):
+        initial = html_escape(self[self._rec_name][0].upper())
```

Comment:
Why the direct access to `_rec_name` instead of `display_name`/`name_get`?

Reply:
The idea is that both `display_name` and `name_get` have various strange overrides, often depending on context, and we don't want the initial or the color depending on the context, we want it stable and unique for one record.

Reply:
Then I think a dedicated hook (probably defaulting to `name` directly. or using a class-level attribute) would make more sense, `_rec_name` is rather odd e.g. it can be an (and often is) an m2o.

Reply:
which option is preferred? creating a hook or a class attribute?

Reply:
They're not exclusive (in fact they're usually both present, with the hook using the class attribute, this allows for easier and more declarative overriding).

But I'd say to just define a hook method to get the "thing to initialize" (or possibly even that thing would return the initial in case e.g. some madperson wants to expand it to *two* letters).

If considered useful, a class attribute can trivially be introduced later on.

Reply:
I think the most declarative approach here, which is also minimal diff, is to just introduce `_avatar_name_field = 'name'` class attribute to replace current usage `_rec_name`.

If we're adding methods for this, we're going back against recent suggestions from TDE and MAT to reduce the number of methods/unused potential overrides.